README.rst

```diff
@@ -0,0 +1,210 @@
+====================
+CMake Common Scripts
+====================
+This contains a set useful cmake macros
+
+.. contents:: Table of Contents
+   :depth: 3
+
+
+Available Macros
+================
+
+Clang Tidy
+----------
+
+This CMake macro will add clang tidy to a provided target.
+
+Macro:
```

Comment:
In general, I think for each macro we should describe what they do and why you want to use them. Having these is really only helpful if everyone knows how to use them (and actually does).

Reply:
Would providing a link to clang-tidy, iwyu, code-coverage and cppcheck documentation be sufficient?

Reply:
I added links to relevant documentation.